src/v/serde/serde.h

```diff
@@ -603,6 +613,10 @@ void read_nested(iobuf_parser& in, T& t, std::size_t const bytes_left_limit) {
     } else if constexpr (is_duration_v<Type>) {
         auto rep = read_nested<int64_t>(in, bytes_left_limit);
         t = std::chrono::duration_cast<Type>(std::chrono::nanoseconds{rep});
+    } else if constexpr (is_time_point_v<Type>) {
+        auto duration = read_nested<typename Type::duration>(
+          in, bytes_left_limit);
+        t = std::chrono::time_point<typename Type::clock>(std::move(duration));
```

Comment:
like std::pair, should we support time point as a native type? imo it is a red flag when i see time points being sent over the wire or stored on disk. i suppose there are legitimate use cases, but maybe those should either be handled explicitly by the caller or we could create a serde::time_point type that forces users to think about what they are doing.

Reply:
You mean it is not logical (no pun intended) due to clock sync issues? (like the general notion of clock drifts)